Tests/Unit/CommonResourceHelper.Tests.ps1

```diff
@@ -0,0 +1,181 @@
+﻿$errorActionPreference = 'Stop'
+Set-StrictMode -Version 'Latest'
+
+Describe 'CommonResourceHelper Unit Tests' {
+    BeforeAll {
+        # Import the CommonResourceHelper module to test
+        $testsFolderFilePath = Split-Path -Path $PSScriptRoot -Parent
+        $moduleRootFilePath = Split-Path -Path $testsFolderFilePath -Parent
+        $dscResourcesFolderFilePath = Join-Path -Path $moduleRootFilePath -ChildPath 'DscResources'
+        $commonResourceHelperFilePath = Join-Path -Path $dscResourcesFolderFilePath -ChildPath 'CommonResourceHelper.psm1'
+        Import-Module -Name $commonResourceHelperFilePath
+    }
+
+    InModuleScope 'CommonResourceHelper' {
+        Describe 'Test-IsNanoServer' {
+            $testComputerInfoNanoServer = @{
+                OsProductType = 'Server'
+                OsServerLevel = 'NanoServer'
+            }
+
+            $testComputerInfoServerNotNano = @{
+                OsProductType = 'Server'
+                OsServerLevel = 'NotNano'
+            }
+
+            $testComputerInfoNotServer = @{
+                OsProductType = 'NotServer'
+                OsServerLevel = 'NotNano'
+            }
+
+            Mock -CommandName 'Test-CommandExists' -MockWith { return $true }
+            Mock -CommandName 'Get-ComputerInfo' -MockWith { return $testComputerInfoNanoServer }
+
+            Context 'Get-ComputerInfo command exists' {
+                Context 'Computer OS type is Server and OS server level is NanoServer' {
+                    It 'Should not throw' {
+                        { $null = Test-IsNanoServer } | Should Not Throw
+                    }
+
+                    It 'Should test if the Get-ComputerInfo command exists' {
+                        $testCommandExistsParameterFilter = {
+                            $nameParameterCorrect = $Name -eq 'Get-ComputerInfo'
+                            return $nameParameterCorrect
+                        }
+
+                        Assert-MockCalled -CommandName 'Test-CommandExists' -ParameterFilter $testCommandExistsParameterFilter -Exactly 1 -Scope 'Context'
+                    }
+
+                    It 'Should retrieve the computer info' {
+                        Assert-MockCalled -CommandName 'Get-ComputerInfo' -Exactly 1 -Scope 'Context'
+                    }
+
+                    It 'Should return true' {
+                        Test-IsNanoServer | Should Be $true
+                    }
+                }
+
+                Mock -CommandName 'Get-ComputerInfo' -MockWith { return $testComputerInfoServerNotNano }
```

Comment:
Not 100% sure what you mean by 'did not clean this up'.

All functions called by the function being tested are mocked at the beginning of the Describe to ensure that no matter what we are expecting the function to do, none of the internal functions are actually called.

Then I declare the Mocks outside the Context because the context doesn't 'contain' the mock. The definition of the mock will continue after the Context even if you declare it within in the Context. This brings up an interesting issue with the new nested Contexts that I tried out here which may be why you are saying the mocks don't look 'cleaned up'. It could be confusing to someone trying to debug a test failure which mock of Get-ComputerInfo will be called in the second Context, but ultimately the function should not be called at all there which we do include a test for.

I'm assuming you would prefer it more like this?
```powershell
Context '' {
   Mock

   It '' {

    }
}
```

rather than
```powershell
Mock

Context '' {
   It '' {

    }
}
```

The first format here required a lot of the exact same Mocks to be redefined repeatedly in most of the resource tests. Though maybe now with the nested Contexts it would make more sense. But then I'm worried that we would have too many levels of nested Contexts in the resource tests. In this tests though, I agree the first format will make more sense.